src/app-layer-enip-common.c

```diff
@@ -0,0 +1,904 @@
+                    return 0;
+                }
+                ENIPExtractUint8(&req_path_instance8, input, &offset);
+                //skip instance, don't need to store
+                bytes_remain--;
+                break;
+            case PATH_ATTR_8BIT: //single attribute
+                if (offset >= (input_len - sizeof(uint8_t)))
+                {
+                    SCLogDebug("DecodeCIPRequestPathPDU: Parsing beyond payload length\n");
+                    return 0;
+                }
+                ENIPExtractUint8(&req_path_attr8, input, &offset);
+                //uint16_t attrib = (uint16_t) req_path_attr8;
+                //SCLogDebug("DecodeCIPRequestPath: 8bit attr 0x%x\n", attrib);
+
+                seg = SCMalloc(sizeof(SegmentEntry));
+                if (unlikely(seg == NULL))
+                    return 0;
+                seg->segment = segment;
+                seg->value = class;
+                TAILQ_INSERT_TAIL(&node->segment_list, seg, next);
+
+                bytes_remain--;
+                break;
+            case PATH_CLASS_16BIT:
+                if (offset >= (input_len - sizeof(uint8_t)))
+                {
+                    SCLogDebug("DecodeCIPRequestPath: Parsing beyond payload length\n");
+                    return 0;
+                }
+                ENIPExtractUint8(&reserved, input, &offset); //skip reserved
+                if (offset >= (input_len - sizeof(uint16_t)))
+                {
+                    SCLogDebug("DecodeCIPRequestPath: Parsing beyond payload length\n");
+                    return 0;
+                }
+                ENIPExtractUint16(&req_path_class16, input, &offset);
+                class = req_path_class16;
+                SCLogDebug("DecodeCIPRequestPath: 16bit class 0x%x\n", class);
+
+                seg = SCMalloc(sizeof(SegmentEntry));
+                if (unlikely(seg == NULL))
+                    return 0;
+                seg->segment = segment;
+                seg->value = class;
+                TAILQ_INSERT_TAIL(&node->segment_list, seg, next);
+
+                bytes_remain = bytes_remain - 2;
+                break;
+            case PATH_INSTANCE_16BIT:
+                if (offset >= (input_len - sizeof(uint8_t)))
+                {
+                    SCLogDebug("DecodeCIPRequestPathPDU: Parsing beyond payload length\n");
+                    return 0;
+                }
+                ENIPExtractUint8(&reserved, input, &offset); // skip reserved
+                if (offset >= (input_len - sizeof(uint16_t)))
+                {
+                    SCLogDebug("DecodeCIPRequestPathPDU: Parsing beyond payload length\n");
+                    return 0;
+                }
+                ENIPExtractUint16(&req_path_instance16, input, &offset);
+                //skip instance, don't need to store
+                bytes_remain = bytes_remain - 2;
```

Comment:
the ' while (bytes_remain > 0)' loop condition implies bytes_remain might be 1 here. Leading to a negative bytes_remain. Would that cause issues?